### PR TITLE
database created

### PR DESCRIPTION
**Pull request by Salam-Dalloul**
database created locally and on heroku

issue  #48

### CODE REVIEW

**Comment by Ghadeer93** [comment]
@Salam-Dalloul  its very good :+1:  , but schema file it must contain test data to help you when you make test 
such as 
``` 
INSERT INTO users  (name, email, password, role) VALUES 
('salam','salam@gmail.com','$2a$08$WsD9bqU5GQdcjk4.eC0JkeCOhdRaVJL5x2gU0OF/vG/pKzeSAExy6','admin');

``` 
so make example for all tables  :+1:

**Review by Ghadeer93 — CHANGES_REQUESTED** [review]
Insert data

File: src/database/build.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const fs = require('fs');
+
+const buildDatabase = () => {
+  const connection = require('./dbConnection');
```
Comment (Ghadeer93) [inline src/database/build.js]: I think you dont need to make function "buildDatabase" :+1:

Comment (Ghadeer93) [inline src/database/build.js]: calling anther file 'dbConnection' it must out of your function

File: src/database/build.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  const sql = fs.readFileSync(`${__dirname}/build.sql`).toString();
```
Comment (Ghadeer93) [inline src/database/build.js]: as same it must out function

File: src/database/build.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  connection.query(sql, (errDb, result) => {
+    if (errDb) {
+      console.log(errDb, 'Error');
+    } else {
+      console.log('Database Created');
+    }
+  });
+};
+buildDatabase();
```
Comment (Ghadeer93) [inline src/database/build.js]: why you calling function ?

File: src/database/build.sql [hunk 4]
```diff
@@ -0,0 +1,51 @@
+BEGIN;
+
```
Comment (Ghadeer93) [inline src/database/build.sql:2]: at first adding this in one line 
```
DROP TABLE IF EXISTS users,shop_owner,cloths,cloths_details,blog,saved CASCADE;
```

File: src/database/build.sql [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+CREATE TABLE IF NOT EXISTS users (
```
Comment (Ghadeer93) [inline src/database/build.sql]: some fields need to change attributes

File: src/database/build.js (hunk 3 above)
Comment (MahmoudMH) [reply src/database/build.js]: i did it

File: src/database/build.js (hunk 2 above)
Comment (MahmoudMH) [reply src/database/build.js]: done

File: src/database/build.js (hunk 1 above)
Comment (MahmoudMH) [reply src/database/build.js]: done

Comment (MahmoudMH) [reply src/database/build.js]: i exported directly and delete the name ..

File: src/database/dbConnection.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+const { Pool } = require('pg');
```
Comment (Ghadeer93) [inline src/database/dbConnection.js:1]: need to import pg module 
```
const pg = require('pg');

```

File: src/database/dbConnection.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+require('env2')('./config.env');
+
+if (!process.env.DATABASE_URL) {
+  throw new Error('DB url not found ');
+} else {
+  module.exports = new Pool({ connectionString: process.env.DATABASE_URL });
```
Comment (Ghadeer93) [inline src/database/dbConnection.js:7]: I think its error.

File: src/database/build.sql (hunk 4 above)
Comment (Salam-Dalloul) [reply src/database/build.sql:2]: i added this for each table!

File: src/database/dbConnection.js (hunk 6 above)
Comment (Salam-Dalloul) [reply src/database/dbConnection.js:1]: this the same as requiring the pg and then require the pool from it!

File: src/database/dbConnection.js (hunk 7 above)
Comment (Salam-Dalloul) [reply src/database/dbConnection.js:7]: if it does not work with you, you need to set up convig.env file! i think! :smile:

**Pull request closed without merging**